pkg/strategy/grid2/twin_order.go

```diff
@@ -111,3 +111,128 @@ func (m TwinOrderMap) String() string {
 	sb.WriteString("================== END OF PIN ORDER MAP ==================\n")
 	return sb.String()
 }
+
+type TwinOrderBook struct {
+	// sort in asc order
+	pins []fixedpoint.Value
+
+	// pin index, use to find the next or last pin in desc order
+	pinIdx map[fixedpoint.Value]int
+
+	// orderbook
+	m map[fixedpoint.Value]*TwinOrder
+
+	size int
+}
+
+func NewTwinOrderBook(pins []Pin) *TwinOrderBook {
```

Comment:
should be newTwinOrderBook since it's not going to be used in other package?

Reply:
It sounds good.